### PR TITLE
added api types (pt2)

### PR DESCRIPTION
**Pull request by wwills2**
feat: added unit schema and typed unit api
feat: added organizations schema and typed organization api

### CODE REVIEW

File: src/renderer/components/blocks/widgets/SyncIndicator.tsx [hunk 1]
```diff
@@ -17,17 +18,19 @@ const SyncIndicator: React.FC<SyncIndicatorProps> = ({ detailed, orgUid }) => {
...
     if (!isSynced) {
-      syncRemaining = organizationList?.reduce((acc, org) => acc + org.sync_remaining, 0) ?? 0;
+      syncRemaining = organizationList?.reduce((acc, org) => {
+        return _.isNil(org.sync_remaining) ? acc : acc + org.sync_remaining;
+      }, 0) ?? 0;
     }
   } else {
     const orgData = organizationsMap?.[orgUid];
-    isSynced = orgData?.synced;
+    isSynced = _.isNil(orgData?.synced) ? false : orgData.synced;
```
Comment (MichaelTaylor3D) [inline src/renderer/components/blocks/widgets/SyncIndicator.tsx:33]: why is this needed? orgData?.synced is falsey

at most you just might need to refactor as `Boolean(orgData?.synced)`

File: src/renderer/schemas/Unit.schema.ts [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+  unitOwner?: string | null;
+  inCountryJurisdictionOfOwner?: string | null;
+  marketplace?: string | null;
+  marketplaceLink?: string | null;
+  marketplaceIdentifier?: string | null;
+  unitTags?: string | null;
+  unitStatusReason?: string | null;
+  updatedAt?: Date | null;
+  createdAt?: Date | null;
+  timeStaged?: number | null; // Assuming the timestamp is represented as a number
```
Comment (MichaelTaylor3D) [inline src/renderer/schemas/Unit.schema.ts:28]: lets remove this, its a weird variable and not consistent or used in the UI. Im worried that if a weird casting issue occurs it could break the app even though the value isnt used.

File: src/renderer/components/blocks/widgets/SyncIndicator.tsx (hunk 1 above)
Comment (wwills2) [reply src/renderer/components/blocks/widgets/SyncIndicator.tsx:33]: good point, resolved
